From 430d1f41ab504e308e9817c152e56913d7ab8de3 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Wed, 2 Oct 2024 11:29:01 +0200
Subject: [PATCH 1257/1695] dts: stm32: add ltdc support on stm32mp23

Add ltdc support on stm32mp23 platform.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1c8f60f998c25457f30ac8a0e3eb7542980ed25f dts: stm32: add ltdc support on stm32mp23]
Change-Id: I7c835f432dc5cdc4418898ded922d9c152154482
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/408340
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp231.dtsi | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index c0c45a3c8..98712cf2b 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -2650,6 +2650,32 @@
 					status = "disabled";
 				};
 			};
+
+			ltdc: display-controller@48010000 {
+				compatible = "st,stm32-ltdc";
+				reg = <0x48010000 0x400>;
+				interrupts = <GIC_SPI 160 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 161 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_BUS_LTDC>, <&rcc CK_KER_LTDC>;
+				clock-names = "bus", "lcd";
+				resets = <&rcc LTDC_R>;
+				status = "disabled";
+				access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_CMN_ID>;
+				access-controller-names = "cmn";
+
+				l1l2 {
+					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_L0L1_ID>;
+					access-controller-names = "l1l2";
+				};
+				l3 {
+					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_L2_ID>;
+					access-controller-names = "l3";
+				};
+				rot {
+					access-controllers = <&rifsc STM32MP25_RIFSC_LTDC_ROT_ID>;
+					access-controller-names = "rot";
+				};
+			};
 		};
 
 		iac: iac@42090000 {
-- 
2.47.2

